source/services/BankidService.js

```diff
@@ -52,7 +52,7 @@ async function collect(orderRef) {
  * @param {string} ssn A Swedish Social Security Number.
  */
 async function auth(ssn) {
-  const endUserIp = await getIPv4Address();
+  const endUserIp = (await getIPv4Address()) || "0.0.0.0";
```

Comment:
How about to default to "0.0.0.0" in `getIPv4Address` function instead of having duplicate code that does it?

Reply:
Please show how :)

Reply:
@dannil76  Check function `getIPV4Address` in path `source/helpers/NetworkInfo.ts`

**Optional change**
```js
export default async function getIPv4Address(): Promise<string> {
  const customNetworkInfoModule: CustomNetworkInfo =
    NativeModules.BasicNetworkInfoModule;

  let ipv4Address;

  if (customNetworkInfoModule) {
    ipv4Address = customNetworkInfoModule.GetIPv4Address();
  } else {
    ipv4Address = NetworkInfo.getIPV4Address();
  }

  return ipv4Address || "0.0.0.0";
}

```

Reply:
It should not be up to `getIPv4Address` to default to "0.0.0.0" as that is a very BankID-specific fallback value. `getIPv4Address` should do one thing only and that is return the IP.

If duplicate code is the concern, would lifting the fallback logic into a separate function work?

```typescript
async function getIPForBankID() {
  const ip = await getIPv4Address();
  return ip ?? "0.0.0.0";
}
```

Reply:
Looks good!